### PR TITLE
Migrate concatenate strings

### PR DESCRIPTION
**Pull request by ngaruko**
# Description

[description]

medic/cht-core #7302 

# Code review checklist

- [ ] Readable: Concise, well named, follows the [style guide](https://docs.communityhealthtoolkit.org/contribute/code/style-guide/), documented if necessary.
- [ ] Documented: Configuration and user documentation on [cht-docs](https://github.com/medic/cht-docs/)
- [ ] Tested: Unit and/or e2e where appropriate
- [ ] Internationalised: All user facing text
- [ ] Backwards compatible: Works with existing data and configuration or includes a migration. Any breaking changes documented in the release notes.

# License

The software is provided under AGPL-3.0. Contributions to this project are accepted under the same license.

### CODE REVIEW

File: tests/page-objects/forms/generic-form.wdio.page.js [hunk 1]
```diff
@@ -8,7 +11,31 @@ const nextPage = async (numberOfPages = 1) => {
   }
 };
 
+const formByHref = (href) => {
```
Comment (dianabarsan) [inline tests/page-objects/forms/generic-form.wdio.page.js:14]: I think this is a little misleading. We're actually passing a form id, not a "href" here. 
I think we should rename this to: `getFormLinkElementById` to be extra clear on what it does.

File: tests/page-objects/forms/generic-form.wdio.page.js [hunk 2]
```diff
@@ -8,7 +11,31 @@ const nextPage = async (numberOfPages = 1) => {
...
+  const css = `.action-container .general-actions .dropup.open .dropdown-menu li a[href="#/reports/add/${href}"]`;
+  return $(css);
+};
+
+const selectForm = async (formId, nonAdminUser = false) => {
```
Comment (dianabarsan) [inline tests/page-objects/forms/generic-form.wdio.page.js:19]: We already have an `openForm` method exposed in reports.wdio.page which we could use instead and not have duplicated logic.

File: tests/page-objects/forms/concatenate-strings.wdio.page.js [hunk 3]
```diff
@@ -50,20 +48,22 @@ const docs = [
       }
     }
   }];
+const submitButton = () => $('[ng-click="onSubmit()"]');
+const reportContent = () =>$('div#reports-content');
```
Comment (dianabarsan) [inline tests/page-objects/forms/concatenate-strings.wdio.page.js:52]: Please use consistent spacing here: 
```suggestion
const reportContent = () => $('div#reports-content');
```

File: tests/e2e/concatenate-xpath-strings.wdio-spec.js [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+  parent: {
+    _id: 'some_parent',
+  },
+};
+
+// If this test starts failing then we need to document in the release notes that we've removed the deprecated
+// feature allowing for concatenation of strings
+describe('Concatenate xpath strings', () => {
+  before(() => loginPage.cookieLogin());
+  beforeEach(() => concatenateStrings.configureForm(userContactDoc));
```
Comment (dianabarsan) [inline tests/e2e/concatenate-xpath-strings.wdio-spec.js:28]: There's only one test, we can just use `before` and not `before` and `beforeEach`.

File: tests/e2e/concatenate-xpath-strings.wdio-spec.js [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+  afterEach(() => utils.revertDb([], true));
```
Comment (dianabarsan) [inline tests/e2e/concatenate-xpath-strings.wdio-spec.js:29]: I believe this is not needed anymore. We already have a global `after` cleanup and there's just one test suite here.

**Review by dianabarsan — CHANGES_REQUESTED** [review]
Nice work! Just some minor observations. 
Also, as part of the Enketo-core upgrade, we will not be supporting this deprecated behavior and this test will be removed: [link to removed file](https://github.com/medic/cht-core/pull/7256/files?authenticity_token=pP89Nv%2FzoIhAUS6MtBKnHeR4MgBgRzw4z6ahdbGtVidurIV2a%2F%2BJ4%2FMrj6fBy5wNdhjkJseAtcnP2RpRqTNSDQ%3D%3D&file-filters%5B%5D=.MD&file-filters%5B%5D=.html&file-filters%5B%5D=.js&file-filters%5B%5D=.json&file-filters%5B%5D=.less&file-filters%5B%5D=.md&file-filters%5B%5D=.patch&file-filters%5B%5D=.png&file-filters%5B%5D=.properties&file-filters%5B%5D=.scss&file-filters%5B%5D=.ts&file-filters%5B%5D=.xlsx&file-filters%5B%5D=.xml&file-filters%5B%5D=.xsl#diff-74df930981a5978977973a95e45af31a0f3a11f90dfa5786982722e29878f25bL1)

**Review by dianabarsan — COMMENTED** [review]
@ngaruko whoops it looks like the merge went wrong and there are 28,000 line changes. Couldyou please fix the merge so only your changes appear in the diff? 
Thanks!

**Comment by ngaruko** [comment]
@dianabarsan some rebasing gone wrong! Fixed that!

**Review by dianabarsan — APPROVED** [review]
Awesome possum

**Pull request merged**